bookkeeper-server/src/main/java/org/apache/bookkeeper/tools/cli/commands/bookies/ReplicasMigrationCommand.java

```diff
@@ -0,0 +1,190 @@
+    }
+    private ReplicasMigrationCommand(ReplicasMigrationFlags flags) {
+        super(CliSpec.<ReplicasMigrationFlags>newBuilder()
+                .withName(NAME)
+                .withDescription(DESC)
+                .withFlags(flags)
+                .build());
+    }
+
+    /**
+     * Flags for replicasMigration command.
+     */
+    @Accessors(fluent = true)
+    @Setter
+    public static class ReplicasMigrationFlags extends CliFlags {
+
+        @Parameter(names = { "-b", "--bookieIds" }, description = "Bookies corresponding to the migrated replica, "
+                + "eg: bookieId1,bookieId2,bookieId3")
+        private String bookieIdsOfMigratedReplica;
+
+        @Parameter(names = { "-l", "--ledgerIds" }, description = "The ledgerIds corresponding to the migrated replica,"
+                + "eg: ledgerId1,ledgerId2,ledgerId3. The default is ALL, which means that all replica data on these "
+                + "bookie nodes will be migrated to other bookie nodes")
+        private String ledgerIdsOfMigratedReplica;
+
+        @Parameter(names = { "-r", "--readOnly" }, description = "Whether to set the bookie nodes of the migrated data"
+                + " to readOnly, the default is true")
+        private boolean readOnly;
+
+        @Parameter(names = {"-p", "--httpPort"}, description = "http port,default is 8080")
+        private String port;
+    }
+
+    @Override
+    public boolean apply(ServerConfiguration conf, ReplicasMigrationFlags cmdFlags) {
+        try {
+            return switchToReadonly(conf, cmdFlags) && replicasMigration(conf, cmdFlags);
+        } catch (Exception e) {
+            throw new UncheckedExecutionException(e.getMessage(), e);
+        }
+    }
+
+    public Set<BookieId> parseBookieIds(String bookieIdsOfMigratedReplica) {
+        Set<BookieId> bookieIdsSet = new HashSet<>();
+        String[] bookieIds = bookieIdsOfMigratedReplica.split(seperator);
+        for (String bookieId : bookieIds) {
+            bookieIdsSet.add(BookieId.parse(bookieId));
+        }
+        return bookieIdsSet;
+    }
+
+    public Set<Long> parseLedgerIds(String ledgerIdsOfMigratedReplica) {
+        Set<Long> ledgerIdSet = new HashSet<>();
+        String[] ledgerIds = ledgerIdsOfMigratedReplica.split(seperator);
+        for (String ledgerId : ledgerIds) {
+            ledgerIdSet.add(Long.parseLong(ledgerId));
+        }
+        return ledgerIdSet;
+    }
+
+    private boolean switchToReadonly(ServerConfiguration conf, ReplicasMigrationFlags flags) {
+        if (flags.readOnly && !conf.isHttpServerEnabled()) {
+            throw new RuntimeException("Please enable http service first, config httpServerEnabled is false!");
+        }
+
+        Set<BookieId> bookieIdSet = parseBookieIds(flags.bookieIdsOfMigratedReplica);
+        for (BookieId bookieId : bookieIdSet) {
+            String bookieAddress = bookieId.getId().split(":")[0];
+            String httpAddress = bookieAddress + ":" + flags.port;
```

Comment:
if the bookie is up and running it publishes on metadata the http endpoint.

Reply:
@eolivelli 
Do you suggest to get the http port of the corresponding bookie from zookeeper?

But sometimes the bookie node may also be down.
So it may be better to let the user input the http port, if you do not enter the http port, use the default port 8080

Reply:
OK , I will fix.

Reply:
fixed.  PTAL,thanks! @eolivelli